x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/BinarySpatialFunction.java

```diff
@@ -182,6 +194,16 @@ protected TypeResolution resolveType(
             if (resolution.unresolved()) {
                 return resolution;
             }
+            // TODO Remove these grid checks once we support geo_shape relation to geoGrid
+            if (resolution.resolved() && DataType.isGeoGrid(spatialExpression.dataType())) {
```

Comment:
Don't we already check for `resolution.unresolved()` above? Looks redundant.

Reply:
OK. I could get rid of one check.